### PR TITLE
Add Quiz constructor to Parser

### PR DESCRIPTION
**Pull request by josephhhhhhhhh**
, move parseChosenOption to the Parser class, add a findTopic function to TopicList and TopicListInterface.

### CODE REVIEW

File: src/main/java/seedu/eduke8/Parser.java [hunk 1]
```diff
@@ -33,15 +47,17 @@ public void parseCommand(String userInput) {
...
-        switch (quizAnswer.trim()) {
-        case "1":
-        case "2":
-        case "3":
-        case "4":
-        default:
-            break;
+    public void parseChosenOption(UiInterface ui, ArrayList<OptionInterface> options, QuestionInterface question) {
+        // Should probably use parser for this part to add hints also
+        int chosen = Integer.parseInt(ui.getInputFromUser());
```
Comment (anqi20) [inline src/main/java/seedu/eduke8/Parser.java]: Would "chosenOption" be a better way to name it?

Comment (josephhhhhhhhh) [reply src/main/java/seedu/eduke8/Parser.java]: Good point! Will make the recommended alteration right away.

File: src/main/java/seedu/eduke8/Parser.java [hunk 2]
```diff
@@ -1,13 +1,27 @@
 package seedu.eduke8;
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/Parser.java:1]: I think it is better if the parser class and its interface could be packaged into their own package, which can be called parser

File: src/main/java/seedu/eduke8/quiz/Quiz.java [hunk 3]
```diff
@@ -39,20 +41,7 @@ private void goThroughQuizQuestions(UiInterface ui, QuestionListInterface questi
                 ui.printOption(options.get(i), i + 1);
             }
 
-            parseChosenOption(ui, options, question);
+            parser.parseChosenOption(ui, options, question);
```
Comment (kstonekuan) [inline src/main/java/seedu/eduke8/quiz/Quiz.java:44]: Thanks for moving this for me

File: src/main/java/seedu/eduke8/Parser.java [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
 
-    public void parseCommand(String userInput) {
+    public Parser() {
+    }
+
+    public void parseCommand(TopicListInterface topicList, String userInput) {
+        int numOfQuestions = 0;
+        String topicName = "";
         String[] commandArr = userInput.trim().split(" ", 0);
+        if (commandArr.length == 3 && commandArr[0].equals("quiz")) {
```
Comment (kstonekuan) [inline src/main/java/seedu/eduke8/Parser.java:21]: Maybe we can move this into the "quiz" case next time so we don't check for the command twice

File: src/main/java/seedu/eduke8/Parser.java [hunk 5]
```diff
@@ -1,13 +1,27 @@
...
+            numOfQuestions = Integer.parseInt(commandArr[2].substring(2));
+            topicName = commandArr[1].substring(2);
+        }
         UiInterface ui = null;
```
Comment (kstonekuan) [inline src/main/java/seedu/eduke8/Parser.java:25]: Remember to initialize this next time as well

**Review by kstonekuan — APPROVED** [review]
LGTM, just keep in mind the minor changes for next time

File: src/main/java/seedu/eduke8/Parser.java [hunk 6]
```diff
@@ -1,13 +1,27 @@
...
+    }
+
+    public void parseCommand(TopicListInterface topicList, String userInput) {
+        int numOfQuestions = 0;
+        String topicName = "";
         String[] commandArr = userInput.trim().split(" ", 0);
+        if (commandArr.length == 3 && commandArr[0].equals("quiz")) {
+            numOfQuestions = Integer.parseInt(commandArr[2].substring(2));
+            topicName = commandArr[1].substring(2);
+        }
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/Parser.java:24]: Great implementation for v1.0! Do take note that we have to handle the case where the input provided by the user is unfavorable though, such as if the input length is not 3.

**Review by Zhi-You — APPROVED** [review]
LGTM

File: src/main/java/seedu/eduke8/Parser.java (hunk 4 above)
Comment (josephhhhhhhhh) [reply src/main/java/seedu/eduke8/Parser.java:21]: Ah yes, pardon my folly, will adjust it in future. Thank you!

File: src/main/java/seedu/eduke8/Parser.java (hunk 5 above)
Comment (josephhhhhhhhh) [reply src/main/java/seedu/eduke8/Parser.java:25]: It was initialised above the if loop.

File: src/main/java/seedu/eduke8/Parser.java (hunk 6 above)
Comment (josephhhhhhhhh) [reply src/main/java/seedu/eduke8/Parser.java:24]: Alright, noted! Will look into how to make the implementation better in future!

File: src/main/java/seedu/eduke8/Parser.java (hunk 5 above)
Comment (kstonekuan) [reply src/main/java/seedu/eduke8/Parser.java:25]: sorry I mean to comment on ui, didn't know the top part got included

**Pull request merged**